atc/radar/resource_scanner.go

```diff
@@ -373,30 +373,44 @@ func (scanner *resourceScanner) check(
 	ctx, cancel := context.WithTimeout(context.Background(), timeout)
 	defer cancel()
 
+	// ChecksDuration time the whole thing
+
 	res := scanner.resourceFactory.NewResourceForContainer(container)
+	checkStartTime := time.Now()
+
 	newVersions, err := res.Check(ctx, source, fromVersion)
 	if err == context.DeadlineExceeded {
 		err = fmt.Errorf("Timed out after %v while checking for new versions - perhaps increase your resource check timeout?", timeout)
 	}
 
+	checkDuration := time.Now().Sub(checkStartTime)
 	resourceConfigScope.SetCheckError(err)
-	metric.ResourceCheck{
-		PipelineName: scanner.dbPipeline.Name(),
-		ResourceName: savedResource.Name(),
-		TeamName:     scanner.dbPipeline.TeamName(),
-		Success:      err == nil,
-	}.Emit(logger)
 
 	if err != nil {
 		if rErr, ok := err.(resource.ErrResourceScriptFailed); ok {
 			logger.Info("check-failed", lager.Data{"exit-status": rErr.ExitStatus})
+			metric.
+				ResourceChecksDuration.
+				WithLabelValues("failed").
```

Comment:
the corresponding example of how you leverage that measurement you defined in `metrics.go`